packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -735,7 +740,13 @@ def _serialize_body_parameter(self, builder: OperationType) -> List[str]:
                 f"_{body_kwarg_name} = self._serialize.body({body_param.client_name}, "
                 f"'{body_param.type.serialization_type}'{is_xml_cmd}{serialization_ctxt_cmd})"
             )
-        elif self.code_model.options["models_mode"] == "dpg":
+        elif self.code_model.options["models_mode"] == "dpg" and (
```

Comment:
this logic seems too confusing and I think it can be simplified a bit. What are we trying to do here?

Reply:
@iscai-msft This function is used by `swagger +DPG` and `swagger + autorest` for overload which is different from "CADL + DPG`. To avoid breaking previous logic, I differentiate it with `hasNativeOverload`